README.md

```diff
@@ -1,346 +1,523 @@
-    hedera-forking-->>-client: Token{}.totalSupply
-
-    client->>-user: Token{}.totalSupply
+You can also include a specific block number.
+For example, the test above is known to work at block `72433403`
+
+```console
+forge test --fork-url https://mainnet.hashio.io/api --fork-block-number 72433403
 ```
 
-The relevant interactions are
+You can use all the tools and cheatcodes Foundry provides, _e.g._, `console.log`
 
-- **(5).** This is the code defined by [HIP-719](https://hips.hedera.com/hip/hip-719#specification).
-  For reference, you can see the
-  [`hedera-services`](https://github.com/hashgraph/hedera-services/blob/fbac99e75c27bf9c70ebc78c5de94a9109ab1851/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/DispatchingEvmFrameState.java#L96)
-  implementation.
-- **(6)**-**(7)**. This calls `getHtsCode` which in turn returns the bytecode compiled from `HtsSystemContract.sol`.
-- **(8)**-**(12)**. This calls `getHtsStorageAt` which uses the `HtsSystemContract`'s [Storage Layout](#storage-layout) to fetch the appropriate state from the Mirror Node. The **(8)** JSON-RPC call is triggered as part of the `redirectForToken(address,bytes)` method call defined in HIP-719.
-  Even if the call from HIP-719 is custom encoded, this method call should support standard ABI encoding as well as defined in
-  [`hedera-services`](https://github.com/hashgraph/hedera-services/blob/b40f81234acceeac302ea2de14135f4c4185c37d/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/common/AbstractCallAttempt.java#L91-L104).
+```solidity examples/foundry-hts/USDCConsole.t.sol
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
 
-## Build
+import {Test, console} from "forge-std/Test.sol";
+import {htsSetup} from "hedera-forking/src/htsSetup.sol";
+import {IERC20} from "hedera-forking/src/IERC20.sol";
 
-This repo consists of two projects/packages.
-A Foundry project, to compile and test the `HtsSystemContract.sol` contract.
-And an `npm` package that implements the `eth_getCode` and `eth_getStorageAt` JSON-RPC calls when HTS emulation is involved.
+contract USDCConsoleExampleTest is Test {
+    function setUp() external {
+        htsSetup();
+    }
 
-To compile the `HtsSystemContract` and test contracts
+    function test_using_console_log() view external {
+        // https://hashscan.io/mainnet/token/0.0.456858
+        address USDC_mainnet = 0x000000000000000000000000000000000006f89a;
 
-```console
-forge build
+        string memory name = IERC20(USDC_mainnet).name();
+        string memory symbol = IERC20(USDC_mainnet).symbol();
+        uint8 decimals = IERC20(USDC_mainnet).decimals();
+        assertEq(name, "USD Coin");
+        assertEq(symbol, "USDC");
+        assertEq(decimals, 6);
+
+        console.log("name: %s, symbol: %s, decimals: %d", name, symbol, decimals);
```

Comment:
nice! Just wanted to make it clear console.log only works for forking network not on regular local network, correct?

Reply:
Love to see this tho!

Reply:
> console.log only works for forking network not on regular local network, correct?

Yes, that's correct.